modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsSouthControllerEpicsAo.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.tcs
+
+import cats.implicits._
+import cats.data.NonEmptySet
+import cats.effect.IO
+import mouse.boolean._
+import monocle.macros.Lenses
+import org.log4s.{Logger, getLogger}
+import seqexec.server.EpicsCodex.encode
+import seqexec.server.SeqexecFailure
+import seqexec.server.gems.Gems
+import seqexec.server.tcs.GemsSource._
+import seqexec.server.tcs.TcsController.{GuiderConfig, ProbeTrackingConfig, Subsystem, wavelengthEq}
+import seqexec.server.tcs.TcsControllerEpicsCommon.{BaseEpicsTcsConfig, GuideControl, agTimeout, applyParam, encodeFollowOption, setGuideProbe, setHrPickup, setNodChopProbeTrackingConfig, setOiwfsProbe, setPwfs1Probe, setScienceFold, setTelescopeOffset, setWavelength, tcsTimeout}
+import seqexec.server.tcs.TcsEpics.VirtualGemsTelescope
+import seqexec.server.tcs.TcsSouthController.{GemsGuiders, TcsSouthAoConfig}
+
+object TcsSouthControllerEpicsAo {
+
+  val Log: Logger = getLogger
+
+  def setNgs1Guide(g: VirtualGemsTelescope,
```

Comment:
There is quite a bit of repeated code here. Would it be difficult to factor this out?